dataland-frontend/tests/e2e/utils/EuTaxonomyFinancialsUpload.ts

```diff
@@ -279,11 +279,27 @@ export function uploadCompanyViaApiAndEuTaxonomyDataForFinancialsViaForm(
 }
 
 /**
- *
+ * Fills the eutaxonomy-financials upload form with the given dataset
  * @param data the data to fill the form with
+ * @param reportingPeriod (optional) to specify reporting period
  */
-export function fillEuTaxonomyForFinancialsRequiredFields(data: EuTaxonomyDataForFinancials): void {
-  cy.wait(4000);
+export function fillAndValidateEuTaxonomyCreditInstitutionForm(
+  data: EuTaxonomyDataForFinancials,
+  reportingPeriod?: string
+): void {
+  if (reportingPeriod && !Number.isNaN(parseInt(reportingPeriod, 10))) {
+    cy.get('[data-test="reportingPeriod"]').click();
+    cy.get(".p-datepicker .p-yearpicker .p-yearpicker-year").each(($element) => {
+      cy.wrap($element)
+        .invoke("text")
+        .then((text) => {
+          if (text.trim() === reportingPeriod.trim()) {
+            cy.wrap($element).click({ force: true });
+          }
+        });
+    });
+  }
+
```

Comment:
I think you can remove this part of the code where you set a specific reportingPeriod, because there is no occurence in the code anymore, where you actually set a specific reportingPeriod value when you use the function `fillAndValidateEuTaxonomyCreditInstitutionForm()`.

Reply:
👍